package-res/resources/web/pentaho/util/promise.js

```diff
@@ -14,27 +14,101 @@
+   *
+   * @name pentaho.util.promise
+   * @namespace
+   * @amd pentaho/util/promise
+   * @ignore
+   */
+  return /** @lends pentaho.util.promise */{
+    /**
+     * Wraps a function call within a promise.
+     *
+     * The function is called **synchronously**.
+     * However, the returned promise is only settled asynchronously.
+     *
+     * The promise is fulfilled with the function's return value
+     * or rejected with an error thrown by the function.
+     *
+     * @param {function() : ?any} fun The function to call.
+     * @param {?Object} [ctx] The object on which to call `fun`.
+     *
+     * @return {Promise.<?any>} A promise for the function's return value.
+     */
+    wrapCall: function(fun, ctx) {
+      if(!fun) throw error.argRequired("fun");
+
       return new Promise(function(resolve) {
         resolve(fun.call(ctx));
       });
     },
 
+    /**
+     * Gets a promise for one or more modules, given their ids.
+     *
+     * When `deps` is a `string`, a single module is required,
+     * and the promise is fulfilled directly with the value of that module.
+     *
+     * When `deps` is an `Array`, zero or more modules are required,
+     * and the promise is fulfilled with an array with the values of those modules.
+     *
+     * Optionally, receives a contextual `require` function,
+     * so that module ids are taken relative to its module's folder.
+     *
+     * @param {!string|!Array.<!string>} deps A single module id or an array of module ids.
+     * @param {?function} [localRequire] A contextual require function.
+     * Defaults to the global `require` function.
+     *
+     * @return {Promise.<any>|Promise.<any[]>} A promise that
+     * is fulfilled with the value of the requested module(s), and
+     * is rejected in case a module loader error occurs (like an undefined module or timeout).
+     */
     require: function(deps, localRequire) {
+      if(deps == null) throw error.argRequired("deps");
+
       var requireFun = localRequire || require;
+
+      if(Array.isArray(deps)) {
+        return new Promise(function(resolve, reject) {
+          requireFun(deps, function() {
+            resolve(arg.slice(arguments));
+          }, reject);
+        });
+      }
+
       return new Promise(function(resolve, reject) {
-        requireFun(deps, resolve, reject);
+        requireFun([deps], resolve, reject);
       });
     },
 
+    /**
+     * Calls a function when a promise settled,
+     * whether it was fulfilled or rejected,
+     * and returns a new, identical promise for others to follow.
+     *
+     * @param {Promise} promise The promise to wait for.
+     * @param {function} fun The function to call.
+     * @param {?object} [ctx] The object on which to call `fun`.
+     *
+     * @return {Promise} An identical promise.
+     */
     "finally": function(promise, fun, ctx) {
+      if(!promise) throw error.argRequired("promise");
```

Comment:
I'm having trouble justifying this function, because it's not much more than a `.then` .
Can you point me to a place it is being used or describe your intentions behind this function?